pkg/elasticsearch/indices.go

```diff
@@ -1,15 +1,70 @@
 package elasticsearch
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
 	"strings"
 
 	"github.com/openshift/elasticsearch-operator/pkg/logger"
 	estypes "github.com/openshift/elasticsearch-operator/pkg/types/elasticsearch"
 	"github.com/openshift/elasticsearch-operator/pkg/utils"
+
+	"github.com/mitchellh/mapstructure"
 )
 
+func (ec *esClient) GetIndex(name string) (*estypes.Index, error) {
+	payload := &EsRequest{
+		Method: http.MethodGet,
+		URI:    name,
+	}
+	logger.DebugObject("GetIndex for %q", name)
+	ec.fnSendEsRequest(ec.cluster, ec.namespace, payload, ec.k8sClient)
+	if payload.Error != nil {
+		return nil, payload.Error
+	}
+	if payload.StatusCode == 404 {
+		return nil, nil
+	}
+	if payload.StatusCode != 200 && payload.StatusCode != 201 {
+		return nil, fmt.Errorf("failed getting index %s. Error code: %v, %v", name, payload.StatusCode != 200, payload.ResponseBody)
+	}
+
+	index := &estypes.Index{}
+	err := mapstructure.Decode(payload.ResponseBody, index)
+	if err != nil {
+		return nil, fmt.Errorf("failed decoding response body into `estypes.Index` for %s: %s", name, err)
+	}
+	index.Name = name
+	return index, nil
+}
+
+func (ec *esClient) CatIndices(name string) (estypes.CatIndicesResponses, error) {
```

Comment:
consider renaming to something like 'GetAllIndices' since 'CatIndices' requires knowledge of ES and the fact that there is a `_cat` API endpoint

Reply:
Done